src/waveform/waveformwidgetfactory.cpp

```diff
@@ -973,3 +1001,22 @@ void WaveformWidgetFactory::startVSync(GuiTick* pGuiTick, VisualsManager* pVisua
 void WaveformWidgetFactory::getAvailableVSyncTypes(QList<QPair<int, QString > >* pList) {
     m_vsyncThread->getAvailableVSyncTypes(pList);
 }
+
+WaveformWidgetType::Type WaveformWidgetFactory::findTypeFromHandleIndex(int index) {
+    WaveformWidgetType::Type type = WaveformWidgetType::Count_WaveformwidgetType;
+    if (index >= 0 && index < (int)m_waveformWidgetHandles.size()) {
+        type = m_waveformWidgetHandles[index].m_type;
+    }
+    return type;
```

Comment:
So this returns an invalid type. Is that case handled in the calling code?

Reply:
No, because there is no caller. Remove?

Reply:
I added it because it is the opposite of the next method, and I was going to use it first on setWidgetTypeFromHandle, but I found that I couldn't express the same if I used the function.

The logic of returning an invalid type instead of the empty type is so that one can detect that it has called the method with an invalid index.